### PR TITLE
fix https://github.com/nitely/nim-regex/issues/24

### PR DESCRIPTION
**Pull request by timotheecour**
we can add `{.deprecated .}` to `toPattern` in subsequent PR

### CODE REVIEW

File: src/regex.nim [hunk 1]
```diff
@@ -2852,12 +2853,15 @@ when defined(forceRegexAtRuntime):
   template re*(s: string): Regex =
     toPattern(s)
 else:
-  template re*(s: string): Regex =
-    ## Parse and compile a regular
-    ## expression at compile-time
+  template re*(s: static string): Regex =
```
Comment (nitely) [inline src/regex.nim]: can we make it a proc? are there pros/cons?

Comment (timotheecour) [reply src/regex.nim]: done; (and nicer tracebacks still hold even with inline)

**Comment by timotheecour** [comment]
PTAL

**Comment by nitely** [comment]
Awesome, thanks so much! LGTM!

**Pull request merged**